### PR TITLE
Support breakpad symbol uploading

### PR DESCRIPTION
**Pull request by kfcampbell**
This change adds support for uploading Breakpad symbols via the `crashes upload-symbols` command. Previously, all types of symbols were uploaded with the symbol_type attribute set to "Apple". 

Now, Breakpad symbols are able to be passed in using the `-b` flag.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "appcenter-cli",
-  "version": "1.1.11",
+  "version": "1.1.12",
```
Comment (owenniblock) [inline package.json]: You don't need to increate the version (we'll do that when we release so that we can roll up any other changes in a single version number). Please can you revert this change?

File: src/commands/crashes/upload-symbols.ts [hunk 2]
```diff
@@ -47,18 +48,30 @@ export default class UploadSymbols extends AppCommand {
   @hasArg
   public sourceMapPath: string;
 
+  @help("Path to a breakpad symbols zip collection.")
```
Comment (owenniblock) [inline src/commands/crashes/upload-symbols.ts]: Will this be a collection of zip files? Or a collection of other files that get zipped? (Apologies - I don't have the context to know this so wanted to check that this help text would be clear to users :) )

File: src/commands/crashes/upload-symbols.ts [hunk 3]
```diff
@@ -68,7 +81,7 @@ export default class UploadSymbols extends AppCommand {
     }
 
     let pathToZipToUpload: string;
-    if (typeof(zip) === "string") {
+    if (typeof (zip) === "string") {
```
Comment (owenniblock) [inline src/commands/crashes/upload-symbols.ts]: Not a biggie but please try and avoid whitespace changes if possible?

File: src/commands/crashes/upload-symbols.ts [hunk 4]
```diff
@@ -158,10 +171,16 @@ export default class UploadSymbols extends AppCommand {
...
+    if (_.isNil(this.symbolsPath) && _.isNil(this.xcarchivePath) && _.isNil(this.breakpadPath)) {
+      throw failure(ErrorCodes.InvalidParameter, "specify either '--symbol', '--xcarchive', or '--breakpad' switch");
     } else if (!_.isNil(this.symbolsPath) && !_.isNil(this.xcarchivePath)) {
       throw failure(ErrorCodes.InvalidParameter, "'--symbol' and '--xcarchive' switches are mutually exclusive");
+    } else if (!_.isNil(this.symbolsPath) && !_.isNil(this.breakpadPath)) {
+      throw failure(ErrorCodes.InvalidParameter, "'--symbol' and '--breakpad' switches are mutually exclusive");
+    } else if (!_.isNil(this.xcarchivePath) && !_.isNil(this.breakpadPath)) {
+      throw failure(ErrorCodes.InvalidParameter, "'--xcarchive' and '--breakpad' switches are mutually exclusive");
+    } else if (!_.isNil(this.breakpadPath) && !_.isNil(this.sourceMapPath)) {
+      throw failure(ErrorCodes.InvalidParameter, "'--breakpad' and '--sourcemap' switches are mutually exclusive");
```
Comment (owenniblock) [inline src/commands/crashes/upload-symbols.ts]: Checking this is true (since `--sourcemap` is not mutually exclusive to any of the other parameters).

File: src/commands/crashes/upload-symbols.ts [hunk 5]
```diff
@@ -182,7 +201,7 @@ export default class UploadSymbols extends AppCommand {
     }
 
     let zipToChange: JsZip;
-    if (typeof(zip) === "string") {
+    if (typeof (zip) === "string") {
```
Comment (owenniblock) [inline src/commands/crashes/upload-symbols.ts]: See whitespace comment above :)

File: test/commands/crashes/upload-symbols/upload-symbols-test.ts [hunk 6]
```diff
@@ -69,10 +69,28 @@ describe("upload-symbols command", () => {
     abortSymbolUploadSpy = Sinon.spy();
   });
 
-  describe("when network requests are successful", () => {
+  describe("when Breakpad network requests are successfu", () => {
```
Comment (owenniblock) [inline test/commands/crashes/upload-symbols/upload-symbols-test.ts]: Small typo in `describe` text (missing `l` off the end)

File: src/commands/crashes/upload-symbols.ts (hunk 2 above)
Comment (kfcampbell) [reply src/commands/crashes/upload-symbols.ts]: done!

File: package.json (hunk 1 above)
Comment (kfcampbell) [reply package.json]: done!

File: src/commands/crashes/upload-symbols.ts (hunk 3 above)
Comment (kfcampbell) [reply src/commands/crashes/upload-symbols.ts]: yeah no problem! this was a remnant of doing an auto-format for this file. i've reverted this and the one below.

File: src/commands/crashes/upload-symbols.ts (hunk 5 above)
Comment (kfcampbell) [reply src/commands/crashes/upload-symbols.ts]: (see my comment above)

File: test/commands/crashes/upload-symbols/upload-symbols-test.ts (hunk 6 above)
Comment (kfcampbell) [reply test/commands/crashes/upload-symbols/upload-symbols-test.ts]: good catch! i don't know how that snuck in there.

File: src/commands/crashes/upload-symbols.ts (hunk 2 above)
Comment (kfcampbell) [reply src/commands/crashes/upload-symbols.ts]: i've changed the wording to "Path to a zip file containing Breakpad symbols." if that's not clear or you'd like me to use different wording, just let me know!

Comment (owenniblock) [reply src/commands/crashes/upload-symbols.ts]: Ah, I think what was confusing me here was the naming `prepareZipFromSymbols` - whereas in this case really we're just sending it a zip file and it's returning the path we send it (which seems a little odd!)

when it's a breakpad - will it always be a .zip? If so - this help text is fine :)

File: src/commands/crashes/upload-symbols.ts (hunk 4 above)
Comment (owenniblock) [reply src/commands/crashes/upload-symbols.ts]: Sorry for not being clearer.

This is what I meant - but I wasn't sure if the check that breakpad and sourcemap are both being passed was correct (and intentional).

File: src/commands/crashes/upload-symbols.ts (hunk 2 above)
Comment (kfcampbell) [reply src/commands/crashes/upload-symbols.ts]: Yep! Breakpad symbols will always be zipped, as mentioned in our documentation here: https://docs.microsoft.com/en-us/appcenter/diagnostics/android-ndk

File: src/commands/crashes/upload-symbols.ts (hunk 4 above)
Comment (kfcampbell) [reply src/commands/crashes/upload-symbols.ts]: Oh I'm sorry, I misunderstood -- that check is actually required. Sourcemaps (as currently implemented in the CLI) only pertain to iOS. Android sourcemaps (to the best of my understanding) would be implemented as "AndroidProGuard". Thus, this check is necessary to make sure the user isn't trying to upload iOS sourcemaps and Android symbols for the same app. I've added it back and resolved this conversation.

If you're think we're good to merge, I think we're good to merge! Just let me know. Otherwise, I'd be happy to discuss this further!
